hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/impl/connector/kafka/StreamKafkaP.java

```diff
@@ -155,17 +171,21 @@ public boolean isCooperative() {
     public boolean saveToSnapshot() {
         if (snapshotTraverser == null) {
             snapshotTraverser = Traversers.traverseIterable(offsets.entrySet())
-                    .onFirstNull(() -> snapshotTraverser = null);
+                                          .map(e -> entry(broadcastKey(e.getKey()), e.getValue()))
+                                          .onFirstNull(() -> snapshotTraverser = null);
         }
         return emitFromTraverserToSnapshot(snapshotTraverser);
```

Comment:
shouldn't the offsets be commited after they have been saved to snapshot?

Reply:
No need to: we don't use them. I commit them when snapshots are disabled to have "approximately once" guarantee.